File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math
import sys
input = sys.stdin.readline





def is_prime(num):
    if num < 2:  
        return 0
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return 0
    return 1



if __name__ == "__main__":
    N = int(input())
    list_num = list(map(int, input().split()))

    result = 0

    for num in list_num:
        result += is_prime(num)
    
    print(result)

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
#include <limits.h>

#define TRUE 1 
#define FALSE 0


int is_prime(int num){
    if(num < 2){
        return FALSE;
    }else if(num == 2){
        return TRUE;
    }

    for(int i = 2; i <= sqrt(num); i++){
        if(num % i == 0){
            return FALSE;
        }
    }

    return TRUE;
}

int main(){
    int N;
    int prime = 0;
    scanf("%d", &N);

    for(int i = 0; i < N; i++){
        int num;
        scanf("%d", &num);

        if(is_prime(num)){
            prime++;
        }
    }

    printf("%d\n", prime);

    return 0;
}